[Feat] Support "proxy" arch between coordinator and prover

noel2004 opened the pull request:
The PR has induced a "coordinator_proxy" app to support the provers cluster mode.

Designation: https://www.notion.so/scrollzkp/Almost-stateless-prover-proxy-2537792d22af80b08db3f8397e0824bd?d=29c7792d22af80ee98df001caf30b7e1#2537792d22af806880efc8cea92d0e19


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Standalone coordinator proxy: proxy server for provers with smarter task routing, per-user session management, priority upstreams, and CLI entrypoint.
  * Proxy API & auth: JWT proxy login flow, per-client sessions, automatic token orchestration, /proxy endpoints for task retrieval and proof submission.

* **Chores**
  * DB migrations, config, Docker/CI workflow, Makefile target, and build artifacts for proxy support.

* **Tests**
  * Expanded proxy and session test coverage.

* **Improvements**
  * Response payload decoding enhanced.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

noel2004 committed:
extend api for proxy

noel2004 committed:
add proxy_login route

noel2004 committed:
WIP: update login logic and coordinator client

noel2004 committed:
extend loginlogic

noel2004 committed:
WIP: controller

noel2004 committed:
WIP: config and client controller

noel2004 committed:
add proxy config sample

noel2004 committed:
WIP: the structure of client manager

noel2004 committed:
framework for auto login

noel2004 committed:
AI step

noel2004 committed:
WIP: AI step

noel2004 committed:
unit test for client

noel2004 committed:
WIP

noel2004 committed:
WIP

noel2004 committed:
WIP

noel2004 committed:
WIP

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
WIP

noel2004 committed:
wip

noel2004 committed:
controller WIP

noel2004 committed:
routes

noel2004 committed:
refining

noel2004 committed:
fix issues from test

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
improve get_task proxy

noel2004 committed:
fix issue in test

noel2004 committed:
fix issues

noel2004 committed:
basic tests

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

lispc committed:
Merge branch 'develop' into coordinator_proxy

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
finish binary

noel2004 committed:
depress link for libzkp

noel2004 committed:
refactor libzkp to be completely mocked out

noel2004 committed:
add compatibile mode and more logs

noel2004 committed:
fix unittest

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
fix config issue

noel2004 committed:
more log for init

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
+ fix get task behavior
+ improve the robust of tests

noel2004 committed:
fix

noel2004 committed:
fix issue

noel2004 committed:
robust prover manager

noel2004 committed:
add persistent for running status

noel2004 committed:
WIP: some refactors

noel2004 committed:
WIP

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
complete persistent layer and unit test

noel2004 committed:
fix concurrent issue

noel2004 committed:
fix goimport issue

noel2004 committed:
add docker action

noel2004 committed:
update config template

noel2004 committed:
Fix login version issue

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
fix after merging

noel2004 committed:
fix issues

noel2004 committed:
refactoring client

noel2004 committed:
cover client reset in test

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
fix issue

noel2004 committed:
continue fixing

noel2004 committed:
fix for edge cases

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
fix security issue

noel2004 committed:
observability: compatible with no db config

noel2004 committed:
Merge remote-tracking branch 'origin/develop' into coordinator_proxy

noel2004 committed:
wip

lispc committed:
Merge branch 'develop' into coordinator_proxy

lispc reviewed (dismissed):
almost fully new codes. approved

Velaciela committed:
fix(rollup): prevent uint64 underflow and nil pointer panic in tests

- Fix gas_oracle/app.go: number-1 underflow when L1 block number is 0
- Fix bridge_test.go: use require.NoError instead of assert.NoError in
  setupEnv/setupDB/prepareContracts to prevent nil pointer panic when
  preceding assertions fail

georgehao commented on coordinator/cmd/proxy/app/app.go:
```diff
@@ -0,0 +1,122 @@
+	apiSrv := server(ctx, cfg, db, registry)
+
+	log.Info(
+		"Start coordinator api successfully.",
```
```suggestion
		"Start coordinator proxy successfully.",
```

Velaciela committed:
docs: add coordinator proxy design document

georgehao committed:
Update coordinator/cmd/proxy/app/app.go
